lint/unexported-call_checker.go

```diff
@@ -0,0 +1,74 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+	"go/types"
+)
+
+type unexportedCallChecker struct {
+	baseLocalExprChecker
+
+	funcName string
+}
+
+func unexportedCallCheck(ctx *context) func(*ast.File) {
+	return wrapLocalExprChecker(&unexportedCallChecker{
+		baseLocalExprChecker: baseLocalExprChecker{ctx: ctx},
+	})
+}
+
+func (c *unexportedCallChecker) PerFuncInit(decl *ast.FuncDecl) bool {
+	if decl.Body == nil {
+		return false
+	}
+	c.funcName = ""
+	cond := decl.Recv != nil &&
+		len(decl.Recv.List) == 1 &&
+		len(decl.Recv.List[0].Names) == 1
+	if cond {
+		c.funcName = decl.Recv.List[0].Names[0].Name
+	}
+	return true
+}
+
+// Check finds calls of unexported method from unexported type
+// outside that type.
+//
+//TODO: update description and warning message
```

Comment:
Please put a spece after `//`.
Comments without a space are treated as "pragmas" in Go gc toolchain.
Especially if "keyword" ends with `:`.